stress_test_model_loan_book.R

```diff
@@ -547,10 +560,29 @@ for (i in seq(1, nrow(transition_scenarios))) {
       year == start_year,
       technology %in% technologies,
       scenario_geography == scenario_geography_filter
+    ) #%>%
+    # distinct(investor_name, portfolio_name, company_name, ald_sector, technology,
+    #          scenario_geography, year, plan_carsten, plan_sec_carsten, term,
+    #          PD_0)
+
+  financial_data_loans_pd <- financial_data_loans %>%
+    select(company_name, company_id, ald_sector, technology, pd) %>%
+    distinct(across(everything()))
```

Comment:
distinct(across(everthing)) is quite problematic. 1) it overshadows data consistency problems and 2) we might still keep duplicates on composite unique columns. How about instead filtering by star_year and then calling report_all_duplicates instead?

Reply:
resolved by filtering on start year before the loop